tests/dashboard_tests.py

```diff
@@ -266,6 +274,157 @@ def test_dashboard_with_created_by_can_be_accessed_by_public_users(self):
 
         assert 'Births' in self.get_resp('/superset/dashboard/births/')
 
+    def test_dashboard_level_access_controls(self):
```

Comment:
Let's break up this test. There's a lot here, and additional test methods would make it clearer what's being tested at each point.